rosidl_generator_cpp/CMakeLists.txt

```diff
@@ -74,7 +74,9 @@ if(BUILD_TESTING)
   endif()
 
   if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Wpedantic)
+    # Disable strict-aliasing checks because test/test_msg_initialization.cpp
+    # has a test that very specifically needs to do aliasing.
+    add_compile_options(-Wall -Wextra -Wpedantic -Wno-strict-aliasing)
```

Comment:
You are right in that GCC Release is the only configuration that causes the warning.  That being said, according to the gcc man page, turning on `-Wall` turns on `-Wstrict-aliasing`, so in theory this should fail for all configurations (with the exception of Windows, of course).  I'm not entirely sure why that isn't happening.

What I'm getting at is that while the only failing case currently is GCC Release, it is actually more correct for us to always disable the strict aliasing check.  This will prevent this problem from cropping up if a newer version of clang (for instance) decides to start enforcing this.